`kamel` run fails for remote files on windows

```
 ./kamel run https://raw.githubusercontent.com/apache/camel-k/b29333f0a878d5d09fb3965be8fe586d77dd95d0/e2e/common/files/yaml.yaml
panic: runtime error: invalid memory address or nil pointer dereference[signal 0xc0000005 code=0x0 addr=0x18 pc=0x19ba09f]
goroutine 1 [running]:github.com/apache/camel-k/pkg/cmd.isLocal(0xc0000b2080, 0x74, 0x1) github.com/apache/camel-k@/pkg/cmd/run.go:653 +0x9fgithub.com/apache/camel-k/pkg/cmd.ResolveSources(0x21fdc40, 0xc0004f4140, 0xc000215c60, 0x1, 0x1, 0xc0003e6200, 0x2, 0x1b1b7a0, 0xc0000a1b80, 0x40c8f0, ...) github.com/apache/camel-k@/pkg/cmd/util_sources.go:68 +0xebgithub.com/apache/camel-k/pkg/cmd.extractModelineOptions(0x21fdc40, 0xc0004f4140, 0xc000215c60, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0) github.com/apache/camel-k@/pkg/cmd/modeline.go:188 +0xa8github.com/apache/camel-k/pkg/cmd.createKamelWithModelineCommand(0x21fdc40, 0xc0004f4140, 0xc00007c090, 0x2, 0x3, 0xc000309920, 0x10, 0x1fbb548, 0x320f9a0, 0x180008, ...) github.com/apache/camel-k@/pkg/cmd/modeline.go:121 +0x3c0github.com/apache/camel-k/pkg/cmd.NewKamelWithModelineCommand(0x21fdc40, 0xc0004f4140, 0xc00007c080, 0x3, 0x4, 0x1fba6d0, 0x0, 0x0, 0xc0002f7b30, 0xc0000a1ef0, ...) github.com/apache/camel-k@/pkg/cmd/modeline.go:64 +0xa2main.main() github.com/apache/camel-k@/cmd/kamel/main.go:45 +0x19e 
```

The same behaviour for `github` prefixed URL

```
./kamel run github:apache/camel-k/e2e/common/files/Java.java./kamel run github:apache/camel-k/e2e/common/files/Java.java
panic: runtime error: invalid memory address or nil pointer dereference[signal 0xc0000005 code=0x0 addr=0x18 pc=0x19ba09f]
goroutine 1 [running]:github.com/apache/camel-k/pkg/cmd.isLocal(0xc000048330, 0x30, 0x1) github.com/apache/camel-k@/pkg/cmd/run.go:653 +0x9fgithub.com/apache/camel-k/pkg/cmd.ResolveSources(0x21fdc40, 0xc000505800, 0xc00057c900, 0x1, 0x1, 0xc00049b200, 0x2, 0x1b1b7a0, 0xc0000a5b80, 0x40c8f0, ...) github.com/apache/camel-k@/pkg/cmd/util_sources.go:68 +0xebgithub.com/apache/camel-k/pkg/cmd.extractModelineOptions(0x21fdc40, 0xc000505800, 0xc00057c900, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0) github.com/apache/camel-k@/pkg/cmd/modeline.go:188 +0xa8github.com/apache/camel-k/pkg/cmd.createKamelWithModelineCommand(0x21fdc40, 0xc000505800, 0xc00007c090, 0x2, 0x3, 0xc00057c060, 0x10, 0x1fbb548, 0x320f9a0, 0x180008, ...) github.com/apache/camel-k@/pkg/cmd/modeline.go:121 +0x3c0github.com/apache/camel-k/pkg/cmd.NewKamelWithModelineCommand(0x21fdc40, 0xc000505800, 0xc00007c080, 0x3, 0x4, 0x1fba6d0, 0x0, 0x0, 0xc000162900, 0xc0000a5ef0, ...) github.com/apache/camel-k@/pkg/cmd/modeline.go:64 +0xa2main.main() github.com/apache/camel-k@/cmd/kamel/main.go:45 +0x19e 
```

--- a/pkg/cmd/run.go
+++ b/pkg/cmd/run.go
@@ -379,7 +379,11 @@ func (o *runCmdOptions) syncIntegration(cmd *cobra.Command, c client.Client, sou
 	files = append(files, o.OpenAPIs...)
 
 	for _, s := range files {
-		if isLocalAndFileExists(s) {
+		ok, err := isLocalAndFileExists(s)
+		if err != nil {
+			return err
+		}
+		if ok {
 			changes, err := sync.File(o.Context, s)
 			if err != nil {
 				return err
@@ -688,12 +692,16 @@ func (o *runCmdOptions) configureTraits(integration *v1.Integration, options []s
 	return nil
 }
 
-func isLocalAndFileExists(fileName string) bool {
+func isLocalAndFileExists(fileName string) (bool, error) {
 	info, err := os.Stat(fileName)
-	if os.IsNotExist(err) {
-		return false
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		// If it is a different error (ie, permission denied) we should report it back
+		return false, errors.Wrap(err, fmt.Sprintf("file system error while looking for %s", fileName))
 	}
-	return !info.IsDir()
+	return !info.IsDir(), nil
 }
 
 func addPropertyFile(fileName string, spec *v1.IntegrationSpec) error {

--- a/pkg/cmd/run_test.go
+++ b/pkg/cmd/run_test.go
@@ -532,7 +532,7 @@ volumes:
 `
 
 	integrationSpec := v1.IntegrationSpec{}
-	err := resolvePodTemplate(context.TODO(),templateText, &integrationSpec)
+	err := resolvePodTemplate(context.TODO(), templateText, &integrationSpec)
 	assert.Nil(t, err)
 	assert.NotNil(t, integrationSpec.PodTemplate)
 	assert.Equal(t, 1, len(integrationSpec.PodTemplate.Spec.Containers))
@@ -543,9 +543,16 @@ func TestResolveJsonPodTemplate(t *testing.T) {
 	integrationSpec := v1.IntegrationSpec{}
 	minifiedYamlTemplate := `{"containers": [{"name": "second"}, {"name": "integration", "env": [{"name": "CAMEL_K_DIGEST", "value": "new_value"}]}]}`
 
-	err := resolvePodTemplate(context.TODO(),minifiedYamlTemplate, &integrationSpec)
+	err := resolvePodTemplate(context.TODO(), minifiedYamlTemplate, &integrationSpec)
 
 	assert.Nil(t, err)
 	assert.NotNil(t, integrationSpec.PodTemplate)
 	assert.Equal(t, 2, len(integrationSpec.PodTemplate.Spec.Containers))
+}
+
+func TestIsLocalFileAndExists(t *testing.T) {
+	value, err := isLocalAndFileExists("/root/test")
+	// must not panic because a permission error
+	assert.NotNil(t, err)
+	assert.False(t, value)
 }

--- a/pkg/cmd/util_content.go
+++ b/pkg/cmd/util_content.go
@@ -30,7 +30,12 @@ func loadRawContent(source string) ([]byte, string, error) {
 	var content []byte
 	var err error
 
-	if isLocalAndFileExists(source) {
+	ok, err := isLocalAndFileExists(source)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if ok {
 		content, err = ioutil.ReadFile(source)
 	} else {
 		u, err := url.Parse(source)

--- a/pkg/cmd/util_sources.go
+++ b/pkg/cmd/util_sources.go
@@ -65,7 +65,12 @@ func ResolveSources(ctx context.Context, locations []string, compress bool) ([]S
 	sources := make([]Source, 0, len(locations))
 
 	for _, location := range locations {
-		if isLocalAndFileExists(location) {
+		ok, err := isLocalAndFileExists(location)
+		if err != nil {
+			return sources, err
+		}
+
+		if ok {
 			answer, err := ResolveLocalSource(location, compress)
 			if err != nil {
 				return sources, err

